### PR TITLE
Yg/all conditions

### PR DESCRIPTION
**Pull request by yoagauthier**
Needs to be rebased after #4 is merged

### CODE REVIEW

**Comment by yoagauthier** [comment]
WIP because I do not evaluate that for some test_var, the condition is also evaluated for not(test_var)

**Comment by SeysT** [comment]
Ready to be reviewed

File: model/criteria.py [hunk 1]
```diff
@@ -20,64 +26,154 @@ def check_criteria_against_paths(self, control_flow_graph, execution_paths):
...
 
     def check_criteria_against_paths(self, control_flow_graph, execution_paths):
-        k_paths = []
+        # all the k paths from the cover graph
+        self.to_cover = control_flow_graph.get_all_k_paths(self.k)
+
         # we get all the possible paths
         for path in execution_paths:
-            k_paths.append(path[:self.k])
+            self.covered.append(path) if len(path) <= self.k else None
```
Comment (yoagauthier) [inline model/criteria.py:95]: If the execution path takes the same nodes than the k_path, but is longer, the k_path should be considered as covered right ? I think it's not the case here.

**Review by yoagauthier — COMMENTED** [review]
Good job for the All-conditions criteria and the coverage %, I did not though of this !

**Pull request merged**